background-repeat: rewrite

estelle opened the pull request:
rewrote to match standard way of writing css property pages.

added description.

removed table of values, converting to definition list

round was explained incorrectly. This has been fixed (images don't get squashed, just stretched)

(the guide in the see also that may be a "broken link" is going to be added to the site when #33461 is merged.)

estelle committed:
Draft: new guide: aspect-ratio

estelle committed:
Merge branch 'main' into ratioguide

estelle committed:
add use cases

estelle committed:
Merge branch 'main' into ratioguide

estelle committed:
current status

estelle committed:
new guide: aspect ratio

estelle committed:
borked repo

estelle committed:
Merge remote-tracking branch 'upstream/main'

estelle committed:
Merge remote-tracking branch 'upstream/main'

estelle committed:
background-repeat: rewrite

bsmth commented on files/en-us/web/css/background-repeat/index.md:
```diff
@@ -11,7 +11,50 @@ The **`background-repeat`** [CSS](/en-US/docs/Web/CSS) property sets how backgro
-By default, the repeated images are clipped to the size of the element, but they can be scaled to fit (using `round`) or evenly distributed from end to end (using `space`).
+## Description
+
+The property accepts two `<repeat-style>` keyterms, or one keyterm as a shorthand for the two values. When two values are provided, the first value defines the horizontal repetition behavior and the second value defines the vertical behavior.
```
Do we define "keyterm" anywhere? It might be nice to explain that.

bsmth commented on files/en-us/web/css/background-repeat/index.md:
```diff
@@ -11,7 +11,50 @@ The **`background-repeat`** [CSS](/en-US/docs/Web/CSS) property sets how backgro
+
+The property accepts two `<repeat-style>` keyterms, or one keyterm as a shorthand for the two values. When two values are provided, the first value defines the horizontal repetition behavior and the second value defines the vertical behavior.
+
+The default value is `repeat repeat`. By default, the background image maintains its intrinsic {{glossary("aspect ratio")}}, repeating both horizontally and vertically to cover the entire background paint area, with edge images being clipped to the size of the element. Property values can be used to repeat only horizontally, vertically, or not at all.
```
>Property values can be used to repeat only horizontally, vertically, or not at all.

Does this mean 'a background image can be repeated either horizontally or vertically'?

bsmth commented on files/en-us/web/css/background-repeat/index.md:
```diff
@@ -11,7 +11,50 @@ The **`background-repeat`** [CSS](/en-US/docs/Web/CSS) property sets how backgro
+
+The default value is `repeat repeat`. By default, the background image maintains its intrinsic {{glossary("aspect ratio")}}, repeating both horizontally and vertically to cover the entire background paint area, with edge images being clipped to the size of the element. Property values can be used to repeat only horizontally, vertically, or not at all.
+
+The repeating images can be evenly spaced apart, ensuring it maintains its aspect ratio without being clipped, with the `space` value. When space is used, if the background paint area has a different aspect ratio than the image or does not otherwise have a size that a multiple of the background size in both directions, there will be areas not covered by the background image.
```
Can we link to 'paint area' if that's described in more detail somewhere?

bsmth commented on files/en-us/web/css/background-repeat/index.md:
```diff
@@ -11,7 +11,50 @@ The **`background-repeat`** [CSS](/en-US/docs/Web/CSS) property sets how backgro
+
+The repeating images can be evenly spaced apart, ensuring it maintains its aspect ratio without being clipped, with the `space` value. When space is used, if the background paint area has a different aspect ratio than the image or does not otherwise have a size that a multiple of the background size in both directions, there will be areas not covered by the background image.
+
+Alternatively, the repeated background image can be scaled to cover the entire area without clipping, by setting the `round` value. This value by lead the background image to be stretched if the aspect ratio of the background image is not the same as the paint area's aspect ratio. With `round`, the repeated images will stretch to fill all the available space until there is room to add an addition ...
```
There's a sentence fragment at:

```md
additional repeated image. When an additional image is able to fit.
```

bsmth replied:
>if an image has a width of `100px`, the image will be repeated 10 times in the horizontal direction when the container is 1000px to 1099px wide, with the image being stretch to 109.9px wide.

I'm not sure I follow this; if the container width is between 1000px and 1100px, then the repeated image will be 100px (at 1000px container width), then stretched until 11 fit at 100px each again (at 1100px container width)?

bsmth commented:
General question after having a read through: what's the starting point of images? Is it top-left 0,0 of an element's position or is there anything else that affects this (directionality / margins etc.)

bsmth commented:
Thanks a lot, @estelle! Looking good, mostly some open questions on some concepts I wasn't sure about!

estelle committed:
rewrite

estelle commented:
> Thanks a lot, @estelle! Looking good, mostly some open questions on some concepts I wasn't sure about!

@bsmth. Thank you for the quick review. I rewrote a lot of it, and think I addressed all your questions.

bsmth commented on files/en-us/web/css/background-repeat/index.md:
```diff
@@ -38,91 +36,79 @@ background-repeat: revert-layer;
+
+When one `<repeat-style>` keyterm is provided, the value is shorthand for the following two-value syntax:
+
+<table class="standard-table">
+  <tbody>
+    <tr>
+      <td><strong>Single value</strong></td>
+      <td><strong>Two-value equivalent</strong></td>
+    </tr>
+    <tr>
+      <td><code>repeat-x</code></td>
+      <td><code>repeat no-repeat</code></td>
+    </tr>
+    <tr>
+      <td><code>repeat-y</code></td>
+      <td><code>no-repeat repeat</code></td>
+    </tr>
+    <tr>
+      <td><code>repeat</code></td>
+      <td><code>repeat repeat</code></td>
+    </tr>
+    <tr>
+      <td><code>space</code></td>
+      <td><code>space space</code></td>
+    </tr>
+    <tr>
+      <td><code>round</code></td>
+      <td><code>round round</code></td>
+    </tr>
+    <tr>
+      <td><code>no-repeat</code></td>
+      <td><code>no-repeat no-repeat</code></td>
+    </tr>
+  </tbody>
+</table>
```
I think we can do md tables for this

```suggestion
| Single value | Two-value equivalent  |
| ------------ | --------------------- |
| `repeat-x`   | `repeat no-repeat`    |
| `repeat-y`   | `no-repeat repeat`    |
| `repeat`     | `repeat repeat`       |
| `space`      | `space space`         |
| `round`      | `round round`         |
| `no-repeat`  | `no-repeat no-repeat` |
```

bsmth reviewed (approved):
Thanks, Estelle. Two formatting suggestions for you but leaving a +1 👍🏻

estelle committed:
Apply suggestions from code review

Co-authored-by: Brian Thomas Smith <brian@smith.berlin>
